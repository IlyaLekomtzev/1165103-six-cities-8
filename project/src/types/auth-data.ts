export type AuthData = {
  email: string;
  password: string;
};

export type UserData = {
  id: number,
  name: string,
  email: string,
  isPro: boolean,
  avatarUrl: string,
  token: string,
};
