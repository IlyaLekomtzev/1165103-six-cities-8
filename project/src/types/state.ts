import { Offer } from './offers';

export type State = {
  city: string;
  sort: string;
  offers: Offer[];
  isLoading: boolean;
  error: string;
};
