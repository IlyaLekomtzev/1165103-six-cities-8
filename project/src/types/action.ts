import {
  ThunkAction,
  ThunkDispatch
} from 'redux-thunk';
import { AxiosInstance } from 'axios';
import { State } from './state';
import * as actions from '../store/action';

export enum ActionType {
  SetCity = 'offers/setCity',
  SetSort = 'offers/setSort',
  SetOffers = 'offers/setOffers',
  SetIsLoading = 'offers/setIsLoading',
  SetError = 'offers/setError',

  SetAuthorizationStatus = 'auth/setAuthorizationStatus',
  SetUserData = 'auth/setUserData',
  RedirectToRoute = 'auth/redirectToRoute',
  RequireLogout = 'auth/requireLogout',
}

type InferValueTypes<T> = T extends {[key: string]: infer U} ? U : never;

export type Actions = ReturnType<InferValueTypes<typeof actions>>;

export type ThunkActionResult<R = Promise<void>> = ThunkAction<R, State, AxiosInstance, Actions>;
export type ThunkAppDispatch = ThunkDispatch<State, AxiosInstance, Actions>;
