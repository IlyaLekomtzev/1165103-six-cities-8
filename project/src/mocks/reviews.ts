import { Review } from '../types/reviews';

export const reviews: Review[] = [
  {
    comment: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.',
    date: '2019-05-08T14:13:56.569Z',
    id: 1,
    rating: 4,
    user: {
      avatarUrl: 'https://images.unsplash.com/photo-1535713875002-d1d0cf377fde?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=880&q=80',
      id: 4,
      isPro: false,
      name: 'Max',
    },
  },
  {
    comment: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. 2',
    date: '2019-06-08T14:13:56.569Z',
    id: 2,
    rating: 3,
    user: {
      avatarUrl: 'https://images.unsplash.com/photo-1535713875002-d1d0cf377fde?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=880&q=80',
      id: 4,
      isPro: false,
      name: 'Max',
    },
  },
  {
    comment: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. 3',
    date: '2019-07-08T14:13:56.569Z',
    id: 3,
    rating: 5,
    user: {
      avatarUrl: 'https://images.unsplash.com/photo-1535713875002-d1d0cf377fde?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=880&q=80',
      id: 4,
      isPro: false,
      name: 'Max',
    },
  },
  {
    comment: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. 4',
    date: '2019-08-08T14:13:56.569Z',
    id: 4,
    rating: 1,
    user: {
      avatarUrl: 'https://images.unsplash.com/photo-1535713875002-d1d0cf377fde?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=880&q=80',
      id: 4,
      isPro: false,
      name: 'Max',
    },
  },
];
