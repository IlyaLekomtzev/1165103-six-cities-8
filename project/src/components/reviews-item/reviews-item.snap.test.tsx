import { render } from '@testing-library/react';
import { Review } from '../../types/reviews';
import ReviewsItem from './reviews-item';

const review: Review = {
  comment: 'Lorem ipsum',
  date: '2021-10-24T07:58:01.572Z',
  id: 1,
  rating: 5,
  user: {
    id: 1,
    name: 'Test',
    isPro: true,
    avatarUrl: 'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=687&q=80',
  },
};

describe('Component: Mistakes', () => {
  it('should render correctly', () => {
    const { container } = render(<ReviewsItem review={review} />);
    expect(container).toMatchSnapshot();
  });
});
