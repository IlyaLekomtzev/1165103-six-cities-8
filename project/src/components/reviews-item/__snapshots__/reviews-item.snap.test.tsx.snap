// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ReviewsItem should render correctly 1`] = `
<div>
  <li
    class="reviews__item"
  >
    <div
      class="reviews__user user"
    >
      <div
        class="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          class="reviews__avatar user__avatar"
          height="54"
          src="https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=687&q=80"
          width="54"
        />
      </div>
      <span
        class="reviews__user-name"
      >
        Test
      </span>
    </div>
    <div
      class="reviews__info"
    >
      <div
        class="reviews__rating rating"
      >
        <div
          class="reviews__stars rating__stars"
        >
          <span
            style="width: 100%;"
          />
          <span
            class="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        class="reviews__text"
      >
        Lorem ipsum
      </p>
      <time
        class="reviews__time"
        datetime="2021-10-24T07:58:01.572Z"
      >
        2021-10-24T07:58:01.572Z
      </time>
    </div>
  </li>
</div>
`;
